# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasource/ContainerParser.h
# Commit: f6df4eb84932
# Full Hash: f6df4eb84932ddaccae4e53d4897234e0153913e
# Author: Chris Pearce <cpearce@mozilla.com>
# Date: 2019-06-14 10:00:22
# Regressor Bug: 1558364
# File Overlap Count: 1
# Description:
#   Bug 1558364 - Add MediaSpan and use it for TrackBuffersManager::mInputBuffer. r=jya
#   
#   As seen in this profile of a Twitch replay: https://perfht.ml/2K9Ydb3 we can
#   often end up spending time in TrackBuffersManager::CodedFrameProcessing()
#   shaving off bytes from the front off TrackBuffersManager::mInputBuffer. This
# ==============================================================================

diff -r e708fe457a95 -r f6df4eb84932 dom/media/mediasource/ContainerParser.h
--- a/dom/media/mediasource/ContainerParser.h	Fri Jun 14 00:37:25 2019 +0000
+++ b/dom/media/mediasource/ContainerParser.h	Fri Jun 14 02:10:09 2019 +0000
@@ -8,6 +8,7 @@
 #define MOZILLA_CONTAINERPARSER_H_
 
 #include "mozilla/RefPtr.h"
+#include "MediaSpan.h"
 #include "MediaContainerType.h"
 #include "MediaResource.h"
 #include "MediaResult.h"
@@ -30,7 +31,7 @@
   // Return NS_OK if segment is present, NS_ERROR_NOT_AVAILABLE if no sufficient
   // data is currently available to make a determination. Any other value
   // indicates an error.
-  virtual MediaResult IsInitSegmentPresent(MediaByteBuffer* aData);
+  virtual MediaResult IsInitSegmentPresent(const MediaSpan& aData);
 
   // Return true if aData starts with a media segment.
   // The base implementation exists only for debug logging and is expected
@@ -38,13 +39,13 @@
   // Return NS_OK if segment is present, NS_ERROR_NOT_AVAILABLE if no sufficient
   // data is currently available to make a determination. Any other value
   // indicates an error.
-  virtual MediaResult IsMediaSegmentPresent(MediaByteBuffer* aData);
+  virtual MediaResult IsMediaSegmentPresent(const MediaSpan& aData);
 
   // Parse aData to extract the start and end frame times from the media
   // segment.  aData may not start on a parser sync boundary.  Return NS_OK
   // if aStart and aEnd have been updated and NS_ERROR_NOT_AVAILABLE otherwise
   // when no error were encountered.
-  virtual MediaResult ParseStartAndEndTimestamps(MediaByteBuffer* aData,
+  virtual MediaResult ParseStartAndEndTimestamps(const MediaSpan& aData,
                                                  int64_t& aStart,
                                                  int64_t& aEnd);
 